apps/distgo/cmd/dist/dist.go

```diff
@@ -29,28 +29,35 @@ import (
 	"github.com/palantir/godel/apps/distgo/cmd"
 	"github.com/palantir/godel/apps/distgo/cmd/build"
 	"github.com/palantir/godel/apps/distgo/params"
-	"github.com/palantir/godel/apps/distgo/pkg/osarch"
 	"github.com/palantir/godel/apps/distgo/pkg/script"
 	"github.com/palantir/godel/apps/distgo/pkg/slsspec"
 )
 
+//
 func Products(products []string, cfg params.Project, forceBuild bool, wd string, stdout io.Writer) error {
-	return build.RunBuildFunc(func(buildSpecWithDeps []params.ProductBuildSpecWithDeps, stdout io.Writer) error {
-		var specsToBuild []params.ProductBuildSpec
-		for _, currSpecWithDeps := range buildSpecWithDeps {
-			if forceBuild {
-				specsToBuild = append(specsToBuild, currSpecWithDeps.AllSpecs()...)
-			} else {
-				specsToBuild = append(specsToBuild, build.RequiresBuild(currSpecWithDeps, nil).Specs()...)
-			}
+	buildSpecsWithDeps, err := build.SpecsWithDepsForArgs(cfg, products, wd)
+	if err != nil {
+		return errors.Wrapf(err, "failed to construct the spec from config")
+	}
+	// build the
+	var specsToBuild []params.ProductBuildSpec
+	for _, currSpecWithDeps := range buildSpecsWithDeps {
+		if forceBuild {
+			specsToBuild = append(specsToBuild, currSpecWithDeps.AllSpecs()...)
+		} else {
+			specsToBuild = append(specsToBuild, build.RequiresBuild(currSpecWithDeps, nil).Specs()...)
 		}
-		if len(specsToBuild) > 0 {
-			if err := build.Run(specsToBuild, nil, build.DefaultContext(), stdout); err != nil {
-				return errors.Wrapf(err, "Failed to build products required for dist")
-			}
+	}
+	if len(specsToBuild) > 0 {
+		if err := build.Run(specsToBuild, nil, build.DefaultContext(), stdout); err != nil {
+			return errors.Wrapf(err, "Failed to build products required for dist")
 		}
-		return cmd.ProcessSerially(Run)(buildSpecWithDeps, stdout)
-	}, cfg, products, wd, stdout)
+	}
+	orderedBuildSpecs, err := Schedule(buildSpecsWithDeps)
+	if err != nil {
+		return errors.Wrapf(err, "")
```

Comment:
nice thing about the `errors` library is that this is not necessary -- if you want to add context to the error, then fill out the message rather than adding the empty string. If you do not need to add extra content, then just return `err` directly (no need to wrap).